HereBeDragons-Pins-1.0.lua

```diff
@@ -276,7 +276,7 @@ local function UpdateMinimapZoom()
 end
 
 local function PositionWorldMapIcon(icon, data, currentMapID, currentMapFloor)
-    local x, y = HBD:GetZoneCoordinatesFromWorld(data.x, data.y, currentMapID, currentMapFloor)
+    local x, y = HBD:GetZoneCoordinatesFromWorld(data.x, data.y, currentMapID, (currentMapID ~= 0 and currentMapFloor or data.instanceID))
```

Comment:
Ah, I didn't think currentMapFloor might not be set. Indeed in that case the order matters.